src/components/config/index.ts

```diff
@@ -63,26 +64,31 @@ export const updateConfig = (data: Partial<Config>) => {
   }
 }
 
-export const setupConfigFromFile = async (path: string) => {
-  const parsed = parseConfig(path)
-
-  if (parsed.symon?.url && parsed.symon?.key) {
+const setupConfig = async (config: Config) => {
+  if (config.symon?.url && config.symon?.key) {
     try {
-      await handshake(parsed)
-
-      // TODO: fetch config from Symon and override local config
-      // Need API in Symon first
+      await handshake(config)
     } catch (error) {
-      log.warn(
-        ` ›   Warning: Can't do handshake with Symon. Monika will use configuration from ${path}.`
-      )
+      log.warn(` ›   Warning: Can't do handshake with Symon.`)
     }
   }
 
-  const validated = validateConfig(parsed)
+  const validated = validateConfig(config)
   if (validated.valid) {
-    cfg = parsed
+    cfg = config
   } else {
     throw new Error(validated.message)
   }
 }
+
+export const setupConfigFromFile = async (path: string) => {
+  const parsed = parseConfig(path)
+
+  await setupConfig(parsed)
+}
+
+export const setupConfigFromUrl = async (url: string) => {
+  const fetched = await fetchConfig(url)
```

Comment:
need to handle if the url is not found for example. right now, the error shown is just

```
 ›   Error: Request failed with status code 404
```

which is not so clear. The message should be

```
The configuration file in <insert_the_url_here> is unreachable. Please check the URL again or your internet connection. 
```

Reply:
done